client/src/components/Form/Elements/FormDialog.vue

```diff
@@ -0,0 +1,68 @@
+<script setup lang="ts">
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faFolderOpen } from "@fortawesome/free-solid-svg-icons";
+import DataDialog from "@/components/DataDialog/DataDialog.vue";
+import { onMounted } from "vue";
+import { getCurrentGalaxyHistory } from "@/utils/data";
+import { reactive } from "vue";
+
+library.add(faFolderOpen);
+
+interface DataDialogProps {
+    id: string;
+    multiple?: boolean;
+    value?: Array<string> | string;
+}
+
+const props = withDefaults(defineProps<DataDialogProps>(), {
+    multiple: false,
+});
+
+const title = "Browse Datasets";
+const historyID: string = "";
+const localHistoryId = reactive({
+    value: historyID,
+});
+
+onMounted(() => {
+    getCurrentGalaxyHistory().then((historyId) => {
+        localHistoryId.value = historyId;
+    });
+});
```

Comment:
This is not reactive. When the current history changes, the component needs to be remounted to reflect this.

If this is not intentional, use `currentHistoryId` from `useHistoryStore`.

Either:

```ts
const historyStore = useHistoryStore();
// ...
historyStore.currentHistoryId; // to access value
```

or

```ts
const { currentHistoryId } = storeToRefs(useHistoryStore());
// ...
currentHistoryId.value; // to access value
```